### PR TITLE
[CI Test] Cronjob overhaul and added sunpydev

### LINKED ISSUE #171: We need a sunpydev CI build

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #171

Clearly did a botch job the first time. 

What milestone? Could be back ported to each version?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,8 +8,8 @@ env:
   global:
     - TOX_ARG=''
 
-name: "Cron jobs"
-if: type = cron
+#name: "Cron jobs"
+#if: type = cron
```
Comment (nabobalis) [inline .travis.yml]: Undo this.

**Comment by nabobalis** [comment]
ndcube currently does fail on SunPy master.

File: tox.ini [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
-envlist = py{36,37}-{offline,astropydev,numpydev,build_docs},conda,pycodestyle
+envlist = py{36,37}-{offline,astropydev,numpydev,sunpydev},build_docs,conda,pycodestyle
 
 [testenv]
-isolated_build=true
+changedir = tmp
 setenv =
     MPLBACKEND = agg
     COLUMNS = 180
+    PYTEST_COMMAND = pytest --pyargs ndcube --cov=ndcube --cov-config={toxinidir}/setup.cfg --verbose --durations=25
```
Comment (Cadair) [inline tox.ini]: We don't need to do pyargs here as there are no C extensions.

Comment (nabobalis) [reply tox.ini]: Is there any harm in leaving it? What if C extensions are added in the future?

Comment (Cadair) [reply tox.ini]: well it means you can't run part of the tests with tox.

Comment (nabobalis) [reply tox.ini]: Updated it.

**Comment by DanRyanIrish** [comment]
Hi @nabobalis.  As far as I can see, this looks good.  Although I do notice that two of the tests are failing.  Is that due to your changes or the fact that there may be incompatibilities of ndcube with sunpy-dev?

**Comment by nabobalis** [comment]
It looks like that ndcube is not compatible with SunPy 1.0 (which sunpydev is testing) at the moment.

**Comment by DanRyanIrish** [comment]
That makes sense, in fairness.  Well then if you an @Cadair are happy with it then I approve it too.

**Pull request merged**